boa/src/builtins/map/mod.rs

```diff
@@ -15,11 +15,92 @@ mod tests;
 #[derive(Debug, Clone)]
 pub(crate) struct Map(OrderedMap<Value, Value>);
 
-impl Map {
-    pub(crate) const NAME: &'static str = "Map";
+impl BuiltIn for Map {
+    const NAME: &'static str = "Map";
+
+    fn attribute() -> Attribute {
+        Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE
+    }
+
+    fn init(context: &mut Context) -> (&'static str, Value, Attribute) {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        let map_object = ConstructorBuilder::new(context, Self::constructor)
+            .name(Self::NAME)
+            .length(Self::LENGTH)
+            .method(Self::set, "set", 2)
+            .method(Self::delete, "delete", 1)
+            .method(Self::get, "get", 1)
+            .method(Self::clear, "clear", 0)
+            .method(Self::has, "has", 1)
+            .method(Self::for_each, "forEach", 1)
+            .callable(false)
+            .build();
+
+        (Self::NAME, map_object.into(), Self::attribute())
+    }
+}
 
+impl Map {
     pub(crate) const LENGTH: usize = 1;
 
+    /// Create a new map
+    pub(crate) fn constructor(this: &Value, args: &[Value], ctx: &mut Context) -> Result<Value> {
+        // Make a new Object which will internally represent the Array (mapping
+        // between indices and values): this creates an Object with no prototype
+
+        // Set Prototype
+        let prototype = ctx.global_object().get_field("Map").get_field(PROTOTYPE);
+
+        this.as_object_mut()
+            .expect("this is array object")
```

Comment:
Should be `"this is map object"`, right?

Reply:
> Should be `"this is map object"`, right?

It was there before i started changing things, I'll change it